cloud/services/compute/loadbalancers/reconcile.go

```diff
@@ -18,72 +18,240 @@ package loadbalancers
 
 import (
 	"context"
+	"errors"
+	"strings"
 
 	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
 	"google.golang.org/api/compute/v1"
-
 	"k8s.io/utils/ptr"
+	infrav1 "sigs.k8s.io/cluster-api-provider-gcp/api/v1beta1"
 	"sigs.k8s.io/cluster-api-provider-gcp/cloud/gcperrors"
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
-// Reconcile reconcile cluster control-plane loadbalancer compoenents.
+// loadBalancingMode describes the load balancing mode that the backend performs.
+type loadBalancingMode string
+
+const (
+	// Utilization determines how the traffic load is spread based on the
+	// utilization of instances.
+	loadBalancingModeUtilization = loadBalancingMode("UTILIZATION")
+
+	// Connection determines how the traffic load is spread based on the
+	// total number of connections that a backend can handle. This is
+	// only mode available for passthrough Load Balancers.
+	loadBalancingModeConnection = loadBalancingMode("CONNECTION")
+
+	loadBalanceTrafficInternal = "INTERNAL"
+)
```

Comment:
Also here let's adhere to the PascalCase conventions

Reply:
These values - `CONNECTION`, `INTERNAL` etc. must match the value that gcp expects. It appears that this is all upper case, from https://cloud.google.com/load-balancing/docs/backend-service#connection_balancing_mode